sympy/core/power.py

```diff
@@ -1696,7 +1700,7 @@ def mul(d1, d2):
         if not (e.is_integer and e.is_positive and (e*d - n).is_nonpositive and
                 res == _mexpand(self)):
             res += O(x**n, x)
-        return res
+        return _mexpand(res)
```

Comment:
This has the undesirable effect of splitting the terms having sums as coefficients into several terms with the same monomial: `(a + b)*x**2  -> a*x**2 + b*x**2`. Would it be possible to apply `_mexpand` earlier for each coefficient separately?

Reply:
It may also be possible to reunite the terms by calling `collect(x)` (either here or possibly at higher level in `series` or `nseries`).

Reply:
I think this would be sufficient - https://github.com/sympy/sympy/blob/fea3004f11f70f62acc0cde5654c0a10ff40fd78/sympy/core/expr.py#L3033 I thought it would be inefficient to call `_mexpand` at an intermediate stage, so I have kept it in the final return.